File: solid-kata/java/lsp-solution/src/main/java/com/codurance/lsp/FillingStation.java
package com.codurance.lsp;

public class FillingStation {

    public void refuel(PetrolCar petrolCar) {
        petrolCar.fillUpWithFuel();
    }

    public void charge(ElectricCar electricCar) {
        electricCar.chargeBattery();
    }
}

File: solid-kata/c#/LSP/FillingStation.cs
namespace LSP
{
    public class FillingStation
    {
        public void Refuel(Vehicle vehicle)
        {
            if (vehicle is PetrolCar)
                vehicle.FillUpWithFuel();
        }

        public void Charge(Vehicle vehicle)
        {
            if (vehicle is ElectricCar)
            {
                vehicle.ChargeBattery();
            }
        }
    }
}